src/Corax/Queries/SortingMatch.cs

```diff
@@ -0,0 +1,325 @@
+                    {
+                        readY = reader.Read<double>(_fieldId, out var value);
+                        iy.Value = (W)(object)new NumericalItem<double>(value);
+                    }
+                    else
+                        throw new NotSupportedException("Not supported.");
+
+                    iy.Match = (long)((ulong)y | (readY ? 0 : NoMatch));
+                }
+
+                if (readX && readY)
+                {
+                    if (typeof(W) == typeof(SequenceItem))
+                    {
+                        return _comparer.CompareSequence(
+                            new ReadOnlySpan<byte>(((SequenceItem)(object)ix.Value).Ptr, ((SequenceItem)(object)ix.Value).Size),
+                            new ReadOnlySpan<byte>(((SequenceItem)(object)iy.Value).Ptr, ((SequenceItem)(object)iy.Value).Size));
+                    }
+                    else if (typeof(W) == typeof(NumericalItem<long>))
+                    {
+                        return _comparer.CompareNumerical(((NumericalItem<long>)(object)ix.Value).Value, ((NumericalItem<long>)(object)iy.Value).Value);
+                    }
+                    else if (typeof(W) == typeof(NumericalItem<double>))
+                    {
+                        return _comparer.CompareNumerical(((NumericalItem<double>)(object)ix.Value).Value, ((NumericalItem<double>)(object)iy.Value).Value);
+                    }
+                }
+                else if (readX)
+                {
+                    return 1;
+                }
+
+                return -1;
+            }
+
+            public void Dispose()
+            {
+                ArrayPool<Item>.Shared.Return(_hash);
+            }
+        }
+
+        public SortingMatch(IndexSearcher searcher, in TInner inner, in TComparer comparer, int take = -1)
+        {
+            _searcher = searcher;
+            _inner = inner;
+            _take = take;
+            _comparer = comparer;
+        }
+
+        public long Count => _inner.Count;
+
+        public QueryCountConfidence Confidence => _inner.Confidence;
+
+        public int AndWith(Span<long> prevMatches)
+        {
+            throw new NotSupportedException($"{nameof(SortingMatch<TInner, TComparer>)} does not support the operation of {nameof(AndWith)}.");
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public int Fill(Span<long> matches)
+{
+            return _comparer.FieldType switch
+            {
+                MatchCompareFieldType.Floating => Fill<NumericalItem<double>>(matches),
+                MatchCompareFieldType.Integer => Fill<NumericalItem<long>>(matches),
+                MatchCompareFieldType.Sequence => Fill<SequenceItem>(matches),
+            };
+        }
+
+        [SkipLocalsInit]
+        private int Fill<W>(Span<long> matches) where W : struct
+        {
+            int take = _take <= 0 ? matches.Length : _take;
```

Comment:
There are scenarios where users will send `0` as the take value. For example, when they just care about the number of results.
In this case, why bother sorting?

Reply:
What happens when Take is greater than Length?

Reply:
That should not be the case by design. If we allow `Take` to be longer that `Length` then we need an entirely different approach.

Reply:
What happens when `Take` is 20K? Or `int.MaxValue` ?

Reply:
For unconstrained `Take` we need a version that would take all the potential values and sort afterwards. Different implementation of Sort. This implementation is constrained, and therefore more performant than the unconstrained.